libraries/DataFlash/DataFlash_File.cpp

```diff
@@ -206,14 +190,13 @@ void DataFlash_File::periodic_fullrate(const uint32_t now)
 
 uint16_t DataFlash_File::bufferspace_available()
 {
-    uint16_t _head;
-    return (BUF_SPACE(_writebuf)) - critical_message_reserved_space();
+    return _writebuf.space() - critical_message_reserved_space();
```

Comment:
When space is less than the reserved, we return negative which wraps to show as positive. Seems this was an existing bug